google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/TableAdminResponses.java

```diff
@@ -0,0 +1,307 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.admin.v2.models;
+
+import static com.google.cloud.bigtable.admin.v2.models.GCRules.GCRULES;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.threeten.bp.Duration;
+import com.google.bigtable.admin.v2.Table.ClusterState.ReplicationState;
+import com.google.bigtable.admin.v2.Table.TimestampGranularity;
+import com.google.api.core.BetaApi;
+import com.google.api.core.InternalApi;
+import com.google.bigtable.admin.v2.CheckConsistencyRequest;
+import com.google.bigtable.admin.v2.GcRule;
+import com.google.bigtable.admin.v2.GcRule.RuleCase;
+import com.google.bigtable.admin.v2.GenerateConsistencyTokenResponse;
+import com.google.bigtable.admin.v2.TableName;
+import com.google.cloud.bigtable.admin.v2.TableAdminClient;
+import com.google.cloud.bigtable.admin.v2.models.GCRules.GCRule;
+import com.google.cloud.bigtable.admin.v2.models.GCRules.IntersectionRule;
+import com.google.cloud.bigtable.admin.v2.models.GCRules.UnionRule;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
+/** Bigtable Table admin response wrappers */
+@BetaApi
+public class TableAdminResponses {
+  private TableAdminResponses() {}
+
+  /**
+   * Converts the protocol buffer table to a simpler Table model with only the required elements
+   *
+   * @param com.google.bigtable.admin.v2.Table - Protobuf table
+   * @return Table - Table response wrapper
+   */
+  @InternalApi
+  public static Table convertTable(com.google.bigtable.admin.v2.Table table) {
+    return new Table(table);
+  }
+
+  /**
+   * Converts the protocol buffer response to a simpler ConsistencyToken which can be passed back as
+   * is.
+   *
+   * @param GenerateConsistencyTokenResponse - Protobuf ConsistencyTokenResponse
+   * @return ConsistencyToken - ConsistencyToken response wrapper
+   */
+  public static ConsistencyToken convertTokenResponse(
+      GenerateConsistencyTokenResponse tokenResponse) {
+    return new ConsistencyToken(tokenResponse);
+  }
+
+  /** Wrapper for {@link Table} protocol buffer object */
+  public static final class Table {
+    private TableName tableName;
+    private TimestampGranularity timestampGranularity;
+    private Map<String, ClusterState> clusterStates = new HashMap<>();
+    private Map<String, ColumnFamily> columnFamilies = new HashMap<>();
+
+    private Table(com.google.bigtable.admin.v2.Table table) {
```

Comment:
please use wrappers in the constructor and move the conversion from protos into a static factory method